Add LusasAdapterV191

peterjamesnugent opened the pull request:
<!-- PLEASE ENSURE YOU REVIEW THE CONTENT OF EACH PR CAREFULLY, INCLUDING SUBSEQUENT COMMENTS BY YOURSELF OR OTHERS. -->
<!-- IN PARTICULAR PLEASE ENSURE THAT SENSITIVE OR INAPPROPRIATE INFORMATION IS NOT UPLOADED -->

### Issues addressed by this PR
<!-- Add reference(s) to issue(s) solved by this PR. Please use keyword Fixes/Closes as per https://help.github.com/articles/closing-issues-using-keywords/ -->

Closes #339

<!-- Add short description of what has been fixed -->


### Test files
<!-- Link to test files to validate the proposed changes -->
[PushPullCompare script](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/16_Test%20Data/Interop%20Test%20Results/Push%20-Pull%20-%20Compare%202.0.gh?csf=1&web=1&e=tsbBa3)

[Save results for LusasV191Adapter here](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/16_Test%20Data/Interop%20Test%20Results/LusasV191Adapter?csf=1&web=1&e=Fhb5mZ
)

[Compare to results for LusasV19Adapter here](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/16_Test%20Data/Interop%20Test%20Results/LusasV19Adapter?csf=1&web=1&e=AClnTG)

### Changelog
<!-- Text to go into changelog if applicable -->
<!-- Please see https://github.com/BHoM/documentation/wiki/changelog for guidelines -->
- Added adapter for LusasV191 
- Changed `type` for `.GetId()` requests from Lusas to `long` to comply with Lusas19.1 updates and previous versions

### Additional comments
<!-- As required -->
You will need to build the Test_Toolkit for the push and pull script to work.

calumlockhart-bh commented:
![image](https://user-images.githubusercontent.com/55385899/186138353-2611812f-8a58-4a12-a3d8-16c90c40e576.png)
There are two different strategies happening here converting to an Int32 (int) or Int64 (long). 
Two Questions:
1)   Would it not make sense to make all the Int32 as Int64'2 instead to future proof?
2)   Would it not make sense to keep everything as Int32, or even reducing them to Int16, in order to reduce the memory usage of the adapter?

peterjamesnugent commented:
Thanks @calumlockhart-bh, appears I missed a few and now updated.

I've added a test script to compare the functionality with the previous versions.

calumlockhart-bh commented on Lusas_Adapter/Convert/ToBHoM/Loads/ToLoadCombination.cs:
```diff
@@ -55,14 +55,15 @@ public static partial class Convert
             {
                 Name = GetName(lusasLoadCombination),
-                Number = lusasLoadCombination.getID(),
+                Number = (int)adapterNameId,
```
As a way to future proof this consider using:

Number = Convert.ToInt32(Math.Max(Math.Min(adapterNameId, int32.MaxValue), int32.MinValue))

This will also stop errors occurring in situations where the long is greater than the (int) can cast to.

peterjamesnugent replied:
Added with warning messages.

calumlockhart-bh commented on Lusas_Adapter/Convert/ToBHoM/Loads/ToLoadcase.cs:
```diff
@@ -35,13 +35,14 @@ public static partial class Convert
             {
                 Name = GetName(lusasLoadcase),
-                Number = lusasLoadcase.getID()
+                Number = (int)adapterNameId,
```
Same as above comment ^

peterjamesnugent replied:
Added with warning message as this will impact combinations that are pulled.

calumlockhart-bh commented:
Works fine in terms of pushing and pulling.

As mentioned above please add: 
Number = Convert.ToInt32(Math.Max(Math.Min(adapterNameId, int32.MaxValue), int32.MinValue))

...or use a simple: 
Number = unchecked((int)adapterNameId), 
which is more efficient but defaults the value to 0 if it exceeds the maximum value for an int.

peterjamesnugent commented:
@BHoMBot check required

calumlockhart-bh reviewed (approved):
Updated suggestion incorporated and happy with functionality from previous test.

peterjamesnugent commented:
@BHoMBot check project-compliance

peterjamesnugent commented:
@BHoMBot check required

peterjamesnugent commented:
@BHoMBot check copyright-compliance

StephennipBH reviewed (approved):
Changes approved. it's been tested using the standard push pull comparer script. Decimal difference are found due to changing to long but this is expected.

peterjamesnugent commented:
@BHoMBot check ready-to-merge
